substrate-archive/src/database.rs

```diff
@@ -75,9 +75,10 @@ pub struct Database {
 impl Database {
 	/// Connect to the database
 	pub async fn new(url: String) -> Result<Self> {
+		let cpus = num_cpus::get().try_into()?;
 		let pool = PgPoolOptions::new()
-			.min_connections(4)
-			.max_connections(28)
+			.min_connections(cpus / 2)
```

Comment:
this may be zero if  `num cpus == 1` probably not very likely but does that cause any issue? 

Maybe do `max(1, cpus / 2)`?

Reply:
…or use `get_physical()` (likely `cpus / 2` anyway)?